tests/data/iterators/bucket_iterator_test.py

```diff
@@ -14,6 +14,22 @@ def test_create_batches_groups_correctly(self):
                                      [self.instances[0], self.instances[1]],
                                      [self.instances[3]]]
 
+    def test_create_batches_groups_correctly_with_max_instances(self):
+        # Because max_instances_in_memory is 3, we bucket the first three,
+        # yield the two batches, then bucket the next 2, and yield that batch.
+        # Since the original order was 4 -> 2 -> 0 -> 1 -> 3,
```

Comment:
This took me a minute to figure out, because I was wondering why we didn't get `[4, 2], [0], [1, 3]`.  Maybe you can add a comment after the order saying `... -> 3, and we take the instances in index order, we should get ...`.

Reply:
comment updated